apps/condo/domains/notification/adapters/smsAdapter.js

```diff
@@ -58,6 +66,122 @@ class SMSAdapter {
+        return AllCountriesTest.test(phoneNumber)
+    }
+
+    async checkIsAvailable () {
+        const result = await fetch(
+            'https://api.instasent.com/organization/account',
+            {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${this.apiKey}`,
+                    'Accept': 'application/json',
+                },
+            }
+        )
+        return result.ok
+    }
+
+    async send ({ phone, message }, extendedParams = {}) {
+        const body = {
+            clientId: crypto.randomBytes(16).toString('hex'),
+            from: 'CondoDevApi',
+            to: phone,
+            text: message,
+            allowUnicode: true,
+            ...extendedParams,
+        }
+        const result = await fetch(
+            'https://api.instasent.com/sms',
+            {
+                method: 'POST',
+                headers: {
+                    'Authorization': `Bearer ${this.apiKey}`,
+                    'Accept': 'application/json',
+                },
+                body: JSON.stringify(body),
+            }
+        )
+        const json = await result.json()
+        const isOk =  result.ok
+        return [isOk, json]
+    }
+}
+
+class TwilioSms {
+
+    isConfigured = false
+
+    constructor () {
+        let config = conf['TWILIO_API_CONFIG'] ?  JSON.parse(conf['TWILIO_API_CONFIG']) : {}
+        this.isConfigured = validateConfig(config, [
+            'accountSid',
+            'authToken',
+            'fromNumber',
+        ])
+        this.accountSid = config.accountSid
+        this.authToken = config.authToken
+        this.fromNumber = config.fromNumber
+    }
+
+    isPhoneSupported (phoneNumber) {
+        return AllCountriesTest.test(phoneNumber)
+    }
+
+    async checkIsAvailable () {
+        const result = await fetch(
+            `https://api.twilio.com/2010-04-01/Accounts/${this.accountSid}/Balance.json`,
+            {
+                method: 'GET',
+                headers: {
+                    'Authorization': 'Basic ' + Buffer.from(this.accountSid + ':' + this.authToken).toString('base64'),
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                },
+            }
+        )
+        return result.ok
+    }
+
+    async send ({ phone, message }, extendedParams = {}) {
+        const body = {
+            To: phone,
+            From: this.fromNumber,
+            Body: message,
+            ...extendedParams,
+        }
+        const result = await fetch(
+            `https://api.twilio.com/2010-04-01/Accounts/${this.accountSid}/Messages.json`,
+            {
+                method: 'POST',
+                headers: {
+                    'Authorization': 'Basic ' + Buffer.from(this.accountSid + ':' + this.authToken).toString('base64'),
+                    'Content-Type': 'application/x-www-form-urlencoded', 
+                },
+                body: Object.entries(body)
```

Comment:
Why are you forming query-url in body? Looks strange

Reply:
I guess you should just be able to pass body object like this:
```suggestion
                body: JSON.stringify(body)
```